ci/upstream.yml

```diff
@@ -29,6 +29,6 @@ dependencies:
   - fsspec
   - pip
   - pip:
-      - icechunk>=0.1.0a12 # Installs zarr-python v3.0.0 as dependency
+      - git+https://github.com/earthmover/icechunk.git@main # Installs zarr-python v3.0.0 as dependency
```

Comment:
Note this PR now uses the `upstream` CI env to actually test against the `main` branch of icechunk